payjoin/tests/integration.rs

```diff
@@ -263,6 +273,189 @@ mod integration {
             Ok(())
         }
 
+        #[tokio::test]
+        async fn test_ns1r() -> Result<(), BoxSendSyncError> {
+            init_tracing();
+            let mut services = TestServices::initialize().await?;
+            tokio::select!(
+            err = services.take_ohttp_relay_handle() => panic!("Ohttp relay exited early: {:?}", err),
+            err = services.take_directory_handle() => panic!("Directory server exited early: {:?}", err),
+            res = test_ns1r(&services) => assert!(res.is_ok(), "NS1R failed: {:#?}", res)
+            );
+
+            async fn test_ns1r(services: &TestServices) -> Result<(), BoxError> {
+                // Anything past 4 senders will hit a max payload size error
+                let num_senders = 4;
+                let (_bitcoind, senders, receiver) =
+                    init_bitcoind_multi_sender_single_reciever(num_senders)?;
+                assert_eq!(senders.len(), num_senders);
+                let reciever_starting_balance = receiver.get_balance(None, None)?;
+
+                struct InnerSenderTestSession {
+                    receiver_session: Receiver,
+                    sender_get_ctx: MultiPartyGetContext,
+                    script_pubkey: ScriptBuf,
+                }
```

Comment:
An `integration::[v2?::]multiparty` mod would give this a more beautiful home.

Reply:
moved to `integration::multiparty`